### PR TITLE
Muon miniAOD DQM

### PR DESCRIPTION
**Pull request by sscruz**

### CODE REVIEW

**Comment by deguio** [comment]
this is apparently still not mergeable.

**Comment by deguio** [comment]
please test

**Comment by sscruz** [comment]
The compilation errors were fixed in the last commit.

**Comment by deguio** [comment]
please test
I am reviewing the code and I have comments. I will post them when I am done with the review.
cheers,
F.

**Comment by folguera** [comment]
the log of the error found is not very informative, I believe it was due to some miscommunication with DAS, could you please have a look and confirm?

**Comment by deguio** [comment]
please test
das error not related.

File: DQMOffline/Configuration/python/DQMOffline_SecondStep_cff.py [hunk 1]
```diff
@@ -112,15 +112,17 @@
                                cscOfflineCollisionsClients *
                                muonQualityTests
                                )
+DQMHarvestMuon_miniAOD = cms.Sequence( muonQualityTests_miniAOD )
```
Comment (deguio) [inline DQMOffline/Configuration/python/DQMOffline_SecondStep_cff.py]: this is not needed. please remove.

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 2]
```diff
@@ -137,3 +138,12 @@
 #miniAOD DQM sequences need to access the filter results.
 PostDQMOfflineMiniAOD = cms.Sequence(miniAODDQMSequence*jetMETDQMOfflineSourceMiniAOD*tracksDQMMiniAOD)
 PostDQMOffline = cms.Sequence()
+
+from DQMOffline.Muon.miniAOD_cff import *                                                                 
+
+muonMiniAOD = cms.Sequence( MuonMiniAOD )
```
Comment (deguio) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: not needed

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 3]
```diff
@@ -137,3 +138,12 @@
...
+DQMOfflineMiniAOD = cms.Sequence( miniAODDQMSequence*muonMonitors_miniAOD*muonMiniAOD )
```
Comment (deguio) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: replace muonMiniAOD with MuonMiniAOD

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 4]
```diff
@@ -112,14 +112,15 @@
...
+DQMOfflineTracking = cms.Sequence( TrackingDQMSourceTier0Common * 
                                    pvMonitor
-                                 )
+                                   )
 DQMOfflineMuon = cms.Sequence( dtSources *
                                rpcTier0Source *
                                cscSources *
                                muonMonitors
                               )
+DQMOfflineMuon_miniAOD = cms.Sequence( muonMonitors_miniAOD )
```
Comment (deguio) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: not needed

File: DQMOffline/Configuration/python/autoDQM.py [hunk 5]
```diff
@@ -4,11 +4,11 @@
             'commonSiStripZeroBias' : ['DQMOfflineCommonSiStripZeroBias',
                                        'PostDQMOffline',
                                        'DQMHarvestCommonSiStripZeroBias+DQMCertCommon'],
-            'trackingOnlyDQM' : ["DQMOfflineTracking",
-                             "DQMHarvestTracking"],
             'muon': ['DQMOfflineMuon',
                      'PostDQMOffline',
                      'DQMHarvestMuon+DQMCertMuon'],
+            'muon_miniAOD': ['DQMOfflineMuon_miniAOD',
```
Comment (deguio) [inline DQMOffline/Configuration/python/autoDQM.py]: no need to edit at all autoDQM.py

File: DQMOffline/Muon/src/MuonMiniAOD.cc [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  else if (WorkingPoint == "mediumMuons")
+    return muon::isMediumMuon(*muon1);
+  else if (WorkingPoint == "looseMuons")
+    return muon::isLooseMuon(*muon1);
+  else if (WorkingPoint == "highPtMuons")
+    return muon::isHighPtMuon(*muon1,thePrimaryVertex);
+  else if (WorkingPoint == "softMuons")
+    return muon::isSoftMuon(*muon1,thePrimaryVertex);
+  else{
+    std::cout << "[E]: MuonMiniAOD. Unknown WP, returning false." << std::endl;
```
Comment (deguio) [inline DQMOffline/Muon/src/MuonMiniAOD.cc]: please make this a logInfo or logWarning

File: DQMOffline/Muon/src/MuonMiniAOD.cc [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  else if (WorkingPoint == "mediumMuons")
+    return muon1 -> isMediumMuon();
+  else if (WorkingPoint == "looseMuons")
+    return muon1 -> isLooseMuon();
+  else if (WorkingPoint == "highPtMuons")
+    return muon1 -> isHighPtMuon(thePrimaryVertex);
+  else if (WorkingPoint == "softMuons")
+    return muon1 -> isSoftMuon(thePrimaryVertex);
+  else{
+    std::cout << "[E]: MuonMiniAOD. Unknown WP, returning false." << std::endl;
```
Comment (deguio) [inline DQMOffline/Muon/src/MuonMiniAOD.cc]: please make this a logInfo or logWarning

**Comment by sscruz** [comment]
I have just addressed the comments.

File: DQMOffline/Configuration/python/DQMOffline_cff.py [hunk 8]
```diff
@@ -137,3 +138,10 @@
 #miniAOD DQM sequences need to access the filter results.
 PostDQMOfflineMiniAOD = cms.Sequence(miniAODDQMSequence*jetMETDQMOfflineSourceMiniAOD*tracksDQMMiniAOD)
 PostDQMOffline = cms.Sequence()
+
+
+DQMOfflineMiniAOD = cms.Sequence( miniAODDQMSequence*muonMonitors_miniAOD*MuonMiniAOD )
```
Comment (deguio) [inline DQMOffline/Configuration/python/DQMOffline_cff.py]: please remove this sequence and place the muon miniAOD in the PostDQMOfflineMiniAOD

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

File: DQMOffline/Muon/python/muonMonitors_cff.py [hunk 9]
```diff
@@ -27,14 +22,14 @@
 
 muonMonitors = cms.Sequence(muonTrackAnalyzers*
                             dtSegmentsMonitor*
-                            cscMonitor*
+#                            cscMonitor*
```
Comment (deguio) [inline DQMOffline/Muon/python/muonMonitors_cff.py]: why this is commented out?

File: DQMOffline/Muon/python/muonMonitors_cff.py [hunk 10]
```diff
@@ -27,14 +22,14 @@
...
                             dqmInfoMuons*
                             muIsoDQM_seq)
-# Modify for if the phase 1 pixel detector is active
-if eras.phase1Pixel.isChosen() :
-    muonMonitors.remove(muonAnalyzer)
+
+muonMonitors_miniAOD = cms.Sequence( muonAnalyzer_miniAOD*
+                                     muIsoDQM_seq_miniAOD)
 
 muonMonitorsAndQualityTests = cms.Sequence(muonMonitors*muonQualityTests)
```
Comment (deguio) [inline DQMOffline/Muon/python/muonMonitors_cff.py]: is this used?

File: DQMOffline/Muon/python/muonMonitors_cff.py [hunk 11]
```diff
@@ -27,14 +22,14 @@
...
+#                            cscMonitor*
                             muonAnalyzer*
                             muonIdDQM*
                             dqmInfoMuons*
                             muIsoDQM_seq)
-# Modify for if the phase 1 pixel detector is active
-if eras.phase1Pixel.isChosen() :
-    muonMonitors.remove(muonAnalyzer)
+
+muonMonitors_miniAOD = cms.Sequence( muonAnalyzer_miniAOD*
```
Comment (deguio) [inline DQMOffline/Muon/python/muonMonitors_cff.py]: is this used?

**Comment by sscruz** [comment]
muonMonitors_miniAOD is indeed needed. It was originally included in the sequence DQMOfflineMiniAOD, which I was asked to erase. I have moved it into PostDQMOfflineMiniAOD.

**Comment by sscruz** [comment]
muonMonitors_miniAOD is indeed needed. It was originally included in the sequence DQMOfflineMiniAOD, which I was asked to erase. I have moved it into PostDQMOfflineMiniAOD.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
+1
provided it passes the tests

File: DQMOffline/Muon/src/MuonMiniAOD.cc [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+
+#include "DataFormats/Common/interface/Handle.h"
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <string>
+#include "TMath.h"
```
Comment (davidlange6) [inline DQMOffline/Muon/src/MuonMiniAOD.cc]: is this include needed?

File: DQMOffline/Muon/src/DiMuonHistograms.cc [hunk 13]
```diff
@@ -180,29 +175,30 @@ void DiMuonHistograms::analyze(const edm::Event & iEvent,const edm::EventSetup&
...
-	    if (iEtaRegion==2) {EtaCutMin= etaECMin;   EtaCutMax=etaECMax;  }
-
-	    if(fabs(recoCombinedGlbTrack1->eta())>EtaCutMin && fabs(recoCombinedGlbTrack1->eta())<EtaCutMax && 
-	       fabs(recoCombinedGlbTrack2->eta())>EtaCutMin && fabs(recoCombinedGlbTrack2->eta())<EtaCutMax){
-	      if (InvMass > 55. && InvMass < 125.) TightTightMuon[iEtaRegion]->Fill(InvMass);
+	  if (charge < 0){
+	    for (unsigned int iEtaRegion=0; iEtaRegion<3; iEtaRegion++){
+	      if (iEtaRegion==0) {EtaCutMin= 0.;         EtaCutMax=2.4;       }
+	      if (iEtaRegion==1) {EtaCutMin= etaBMin;    EtaCutMax=etaBMax;   } 
+	      if (iEtaRegion==2) {EtaCutMin= etaECMin;   EtaCutMax=etaECMax;  }
```
Comment (davidlange6) [inline DQMOffline/Muon/src/DiMuonHistograms.cc]: could these if statements go into a place where they get executed once? (e.g., just store them into an array that you then loop over)
@sscruz

**Comment by deguio** [comment]
please test

**Comment by sscruz** [comment]
Hi, I believe this error is not related to this PR. Can someone confirm this?

**Comment by deguio** [comment]
please test

File: DQMOffline/Muon/src/DiMuonHistograms.cc [hunk 14]
```diff
@@ -67,42 +68,33 @@ void DiMuonHistograms::bookHistograms(DQMStore::IBooker & ibooker,
 				      edm::EventSetup const & /* iSetup */){
   
   ibooker.cd();
-  ibooker.setCurrentFolder("Muons/DiMuonHistograms");  
+  ibooker.setCurrentFolder(theFolder);  
 
-  int nBin = 0, nBinLM = 0;
+  int nBin[3] = {etaBin,etaBBin,etaEBin};
+  EtaName[0]  = ""; EtaName[1] = "_Barrel"; EtaName[3] = "_EndCap";
```
Comment (deguio) [inline DQMOffline/Muon/src/DiMuonHistograms.cc]: EtaName[3] = "_EndCap";
should be
EtaName[2] = "_EndCap";

**Comment by deguio** [comment]
the last commit is buggy..  :(
F.

**Comment by folguera** [comment]
sorry for the mess there... we hope now it is fixed and satisfying all the comments from you.

**Comment by deguio** [comment]
ciao,
I believe that yes, but the branch is nor mergeable anymore. could you please rebase?
thanks,
F.

**Comment by deguio** [comment]
still not mergeable

**Comment by folguera** [comment]
@deguio I think we fixed the issue, could you please trigger the tests?  Thanks and sorry for the inconvenience...

**Comment by deguio** [comment]
please test

**Comment by sscruz** [comment]
@deguio I think that the compilation error is not related to this PR.

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
I think there is still a comment by david to be addressed, but the code looks in good shape to me.

**Comment by folguera** [comment]
Sorry for this, do you prefer that we address this comment now or in a different PR?

**Comment by sscruz** [comment]
Hi,  I've been working on completely addressing David's comment. The code that I've produced compiles correctly.  However, I've been finding some errors when trying to test it, not related to the modules that I'm modifying, but that don't allow me to test them. When running  [1], I get [2]. Any hint in which can be the cause? I'm using CMSSW_8_0_0_pre5. 
Cheers and many thanks.

[1]  cmsDriver.py test -s DQM:@miniAODDQM -n 30 --eventcontent DQM --conditions auto:com10 --geometry Ideal --mc --filein /store/relval/CMSSW_8_0_0_pre5/RelValZMM_13/MINIAODSIM/PUpmx25ns_80X_mcRun2_asymptotic_v1-v1/00000/D0E4D3A1-2CC6-E511-B5D6-0025905A60EE.root

[2] DQM:@miniAODDQM,ENDJOB
entry /store/relval/CMSSW_8_0_0_pre5/RelValZMM_13/MINIAODSIM/PUpmx25ns_80X_mcRun2_asymptotic_v1-v1/00000/D0E4D3A1-2CC6-E511-B5D6-0025905A60EE.root
 ################################################################### 
 # WARNING: this module is deprecated.                             # 
 # Please use CondCore.CondDB.CondDB_cfi.py                        # 
 ################################################################### 
Step: DQM Spec: ['@miniAODDQM']
Step: ENDJOB Spec: 
# Conditions read from  CMS_CONDITIONS  via FrontierProd 
Starting  cmsRun  test_DQM.py
 ################################################################### 
 # WARNING: this module is deprecated.                             # 
 # Please use CondCore.CondDB.CondDB_cfi.py                        # 
 ################################################################### 
# Conditions read from  CMS_CONDITIONS  via FrontierProd 
----- Begin Fatal Exception 10-Feb-2016 18:15:34 CET-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
Exception Message:
unable to load /afs/cern.ch/work/s/sesanche/private/DQM/Feb10/CMSSW_8_0_0_pre6/lib/slc6_amd64_gcc530/poisoned/plugin-poisoned-L1TriggerL1TCalorimeterPlugins.so because /afs/cern.ch/work/s/sesanche/private/DQM/Feb10/CMSSW_8_0_0_pre6/lib/slc6_amd64_gcc530/poisoned/plugin-poisoned-L1TriggerL1TCalorimeterPlugins.so: cannot open shared object file: No such file or directory
----- End Fatal Exception -------------------------------------------------

**Comment by deguio** [comment]
the errors don't seem to be related. could you try with a more recent IB as a starting point?

**Comment by sscruz** [comment]
Hi, I've tried with a couple of IBs: CMSSW_8_0_X_2016-02-01-2300, CMSSW_8_0_X_2016-02-11-2300, and also with CMSSW_8_0_0_pre5, but all of them deliver a similar error.

**Comment by deguio** [comment]
I don't see errors in the last automatic test

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
forgot to mention that a last rebase is needed apparently.
thanks!
F.

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
need to have 81x merged first

**Comment by slava77** [comment]
-1
looks like the PR is based on 81X, while it is instead submitted to 80X

**Comment by sscruz** [comment]
Hi, sorry for the inconvenience. I think I misunderstood @deguio 's last comment. Do I have to rebase for 80X or 81X?

**Comment by slava77** [comment]
https://github.com/cms-sw/cmssw/pull/12996#issuecomment-208940479
said ``` need to have 81x merged first ```
nothing about a rebase, all new feature PRs should be merged to 81X (development release) first and then proposed to merge to a production release like 80X.

**Comment by deguio** [comment]
-1

**Comment by davidlange6** [comment]
i'm closing this - lets sort it out in 81x.

**Pull request closed without merging**